onnxruntime/python/onnxruntime_inference_collection.py

```diff
@@ -381,15 +381,18 @@ def __init__(
 
         try:
             self._create_inference_session(providers, provider_options, disabled_optimizers)
-        except ValueError as e:
+        except (ValueError, RuntimeError) as e:
             if self._enable_fallback:
                 print(f"EP Error {e} when using {providers}")
                 print(f"Falling back to {self._fallback_providers} and retrying.")
                 self._create_inference_session(self._fallback_providers, None)
                 # Fallback only once.
                 self.disable_fallback()
                 return
-            raise
+            raise RuntimeError(
```

Comment:
in case fallback is not enabled, should we just raise the original `ValueError` or `RuntimeError`?